src/easydiffraction/utils/decorators.py

```diff
@@ -2,25 +2,44 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
 import inspect
+import typing
+
+import numpy as np
 
 
 def enforce_type(func):
     sig = inspect.signature(func)
-    params = list(sig.parameters.items())
 
-    # Find the first argument that isn't 'self'
-    for name, param in params:
-        if name != 'self':
-            expected_type = param.annotation
+    def wrapper(self, *args, **kwargs):
```

Comment:
This is a good step towards enforcing types. However, there's no `Union` or `Optional` support nor does it handle default values well.

Maybe just reuse existing modules like [typeguard](https://github.com/agronholm/typeguard)

Reply:
Right, this was just a temporary solution. I've already mentioned alternatives such as typeguard in the dedicated issue: https://github.com/easyscience/diffraction-lib/issues/87#issuecomment-3248855534